### PR TITLE
Limit event.timestamp for event_monitoring_live

### PR DESCRIPTION
**Pull request by scholtzan**
I'm seeing some materialized views that are not getting updated due to issues like these:
`Adding 256371183196959 MILLISECOND to timestamp 2023-11-25 20:44:00.000000+00 causes overflow`

Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.
- [ ] When adding a new derived dataset, ensure that data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data can be available in the [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](https://github.com/mozilla/bigquery-etl/blob/main/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/DENG-2172)

### CODE REVIEW

File: sql_generators/glean_usage/templates/event_monitoring_live_v1.init.sql [hunk 1]
```diff
@@ -10,7 +10,8 @@ IF
         TIMESTAMP_TRUNC(
           TIMESTAMP_ADD(
             SAFE.PARSE_TIMESTAMP('%FT%H:%M%Ez', ping_info.start_time),
-            INTERVAL event.timestamp MILLISECOND
+            -- limit event.timestamp, otherwise this will cause an overflow
+            INTERVAL LEAST(event.timestamp, 200000000000000) MILLISECOND
```
Comment (scholtzan) [inline sql_generators/glean_usage/templates/event_monitoring_live_v1.init.sql:14]: I've tested 200000000000000 as limit and it didn't break the query

**Review by akkomar — APPROVED** [review]
LGTM as an overflow fix.

I'd consider using some smaller value (current one equals ~6k years)
There's also `ping_info.end_time` that theoretically should be an upper bound but it also could be botched.

File: sql_generators/glean_usage/templates/event_monitoring_live_v1.init.sql [hunk 2]
```diff
@@ -7,47 +7,21 @@ IF
     SELECT
       DATE(submission_timestamp) AS submission_date,
       TIMESTAMP_ADD(
-        TIMESTAMP_TRUNC(
-          TIMESTAMP_ADD(
-            SAFE.PARSE_TIMESTAMP('%FT%H:%M%Ez', ping_info.start_time),
-            INTERVAL event.timestamp MILLISECOND
-          ),
-          HOUR
-        ),
```
Comment (scholtzan) [inline sql_generators/glean_usage/templates/event_monitoring_live_v1.init.sql:16]: Use `ping_info.start_time` only, just like we do in the aggregate table

Comment (scholtzan) [reply sql_generators/glean_usage/templates/event_monitoring_live_v1.init.sql:16]: alright changing to submission_timestamp

**Pull request merged**